flags.cmake

```diff
@@ -0,0 +1,201 @@
+endforeach()
+
+foreach(flag IN LISTS maybe_flags)
+    set(stripped_flag "${flag}")
+    if(flag MATCHES ".<.<COMPILE_LANGUAGE:[A-Z]+>:(.*)>")
+        set(stripped_flag "${CMAKE_MATCH_1}")
+    endif()
+    string(MAKE_C_IDENTIFIER "HAVE_FLAG_${stripped_flag}" var)
+    set(CMAKE_REQUIRED_QUIET TRUE)
+    set(first_check FALSE)
+    if(NOT DEFINED "${var}")
+        set(first_check TRUE)
+        set(msg "Checking support for compile flag: ${stripped_flag}")
+        message(STATUS "${msg}")
+    endif()
+    check_cxx_compiler_flag("${stripped_flag}" "${var}")
+    if(first_check)
+        if(NOT "${${var}}")
+            message(STATUS "${msg} - No")
+        else()
+            message(STATUS "${msg} - Yes")
+        endif()
+    endif()
+    if("${${var}}")
+        target_compile_options(openrct2-flags-iface INTERFACE ${flag})
+    endif()
+endforeach()
+
+cmake_pop_check_state()
+
+if(NOT ENABLE_RCT2)
+    target_compile_definitions(openrct2-flags-iface INTERFACE NO_RCT2)
+endif()
+if(ENABLE_LIGHTFX)
+    target_compile_definitions(openrct2-flags-iface INTERFACE __ENABLE_LIGHTFX__)
+endif()
+if(NOT ENABLE_TTF)
+    target_compile_definitions(openrct2-flags-iface INTERFACE NO_TTF)
+endif()
+if(NOT ENABLE_HTTP_TWITCH)
+    target_compile_definitions(openrct2-flags-iface INTERFACE DISABLE_HTTP DISABLE_TWITCH)
+endif()
+if(NOT ENABLE_NETWORK)
+    target_compile_definitions(openrct2-flags-iface INTERFACE DISABLE_NETWORK)
+endif()
+if(USE_MMAP_FOR_RCT2)
+    target_compile_definitions(openrct2-flags-iface INTERFACE USE_MMAP)
+endif()
+
+if(MSVC)
+    target_compile_definitions(openrct2-flags-iface INTERFACE
+        _CRT_SECURE_NO_WARNINGS
+        _USE_MATH_DEFINES
+        _WINSOCK_DEPRECATED_NO_WARNINGS
+        NOMINMAX
+        CURL_STATICLIB
+        ZIP_STATIC
+        SDL_MAIN_HANDLED
+        )
+    target_link_libraries(openrct2-flags-iface INTERFACE
+        -opt:nolbr
+        -ignore:4099
+        imm32.lib
+        version.lib
+        winmm.lib
+        crypt32.lib
+        wldap32.lib
+        )
+endif()
+
+# Set the debug level macro
+target_compile_definitions(openrct2-flags-iface INTERFACE DEBUG=${DEBUG_LEVEL})
+
+# Turn on PIC
+set_property(TARGET openrct2-flags-iface PROPERTY INTERFACE_POSITION_INDEPENDENT_CODE TRUE)
+
+if(NOT MSVC)
+    # Ensure we have compiler version flags set
+    # (CMake has trouble with the meta-features on MSVC)
+    set(comp_features
+        c_std_11 # C code should be compiled as C11
+        cxx_std_14 # C++ code as C++14
+        )
+    set_target_properties(openrct2-flags-iface PROPERTIES
+        INTERFACE_COMPILE_FEATURES "${comp_features}"
+        )
+endif()
+
+if(APPLE)
+    target_compile_options(openrct2-flags-iface INTERFACE -Wno-error=objc-method-access)
+endif()
+
+# Enable dlopen on Unix systems
+if(UNIX AND NOT CMAKE_SYSTEM_NAME MATCHES BSD)
+    target_link_libraries(openrct2-flags-iface INTERFACE dl)
+endif()
+
+# Enable multithreading
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
```

Comment:
So far we only need threads for HTTP request. There are platforms without threading support that OpenRCT2 handles (e.g. emscripten), so this needs to be a compile-time option.

Reply:
Fixed